### PR TITLE
Refactor/cda migration

### PR DESCRIPTION
**Pull request by radireddy**
| Question         | Answer
| ---------------- | ---
| Branch?          | develop
| Bug fix?         | no
| New feature?     | no
| Breaking change? | no
| Tickets          | Fix #MANAGER-4014
| License          | BSD 3-Clause

## Description

1. Created new standalone module and app for CDA
2. MIGRATED CDA CODE FROM cloud to the above module.
3. Integrated above standalone module/app in dedicated.

### CODE REVIEW

**Comment by cbourgois** [comment]
Some `--signoff` missing in commits.

**Comment by cbourgois** [comment]
Could you add redirection in cloud to help customer who have bookmarks to be redirected to dedicated ? Thanks :)
https://github.com/ovh/manager/blob/master/packages/manager/apps/cloud/client/app/app.js#L112-L118

File: packages/manager/modules/cda/src/ip/list/cda-ip-list.controller.js [hunk 1]
```diff
@@ -1,6 +1,9 @@
-angular
-  .module('managerApp')
-  .controller('CdaIpListCtrl', function CdaIpListCtrl(
+import IpAddTemplate from '../add/cda-ip-add.html';
+import IpDeleteTemplate from '../delete/cda-ip-delete.html';
```
Comment (cbourgois) [inline packages/manager/modules/cda/src/ip/list/cda-ip-list.controller.js]:
```suggestion
import ipAddTemplate from '../add/cda-ip-add.html';
import ipDeleteTemplate from '../delete/cda-ip-delete.html';
```

File: packages/manager/apps/cda/package.json [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    "dev": "webpack-dev-server",
+    "dev:watch": "yarn run dev",
+    "start": "lerna exec --stream --scope='@ovh-ux/manager-cda-app' --include-dependencies -- npm run build --if-present",
+    "start:dev": "lerna exec --stream --scope='@ovh-ux/manager-cda-app' --include-dependencies -- npm run dev --if-present",
+    "start:watch": "lerna exec --stream --parallel --scope='@ovh-ux/manager-cda-app' --include-dependencies -- npm run dev:watch --if-present"
+  },
+  "dependencies": {
+    "@ovh-ux/manager-config": "^0.4.0",
+    "@ovh-ux/manager-core": "^8.0.0",
+    "@ovh-ux/manager-iplb": "^0.0.0 || ^0.1.0",
```
Comment (marie-j) [inline packages/manager/apps/cda/package.json]: If the module has been released maybe we can 
```suggestion
    "@ovh-ux/manager-iplb": "^0.1.0",
```

File: packages/manager/apps/cda/package.json [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    "build:us": "webpack --env.production --env.region='US' && mv dist dist-US",
+    "dev": "webpack-dev-server",
+    "dev:watch": "yarn run dev",
+    "start": "lerna exec --stream --scope='@ovh-ux/manager-cda-app' --include-dependencies -- npm run build --if-present",
+    "start:dev": "lerna exec --stream --scope='@ovh-ux/manager-cda-app' --include-dependencies -- npm run dev --if-present",
+    "start:watch": "lerna exec --stream --parallel --scope='@ovh-ux/manager-cda-app' --include-dependencies -- npm run dev:watch --if-present"
+  },
+  "dependencies": {
+    "@ovh-ux/manager-config": "^0.4.0",
+    "@ovh-ux/manager-core": "^8.0.0",
```
Comment (marie-j) [inline packages/manager/apps/cda/package.json]:
```suggestion
    "@ovh-ux/manager-core": "^8.0.0 || ^9.0.0",
```
some breaking changes are coming to `@ovh-ux/manager-core`

Comment (radireddy) [reply packages/manager/apps/cda/package.json:26]: should we take care of these breaking changes as part of repsac migration?

**Comment by ovh-cds** [comment]
CDS Report installation#11261.0 ✘  
* Stage 1
  * Install ✘

**Comment by ovh-cds** [comment]
CDS Report installation#11262.0 ✘  
* Stage 1
  * Install ✘

File: packages/manager/apps/cda/package.json (hunk 3 above)
Comment (marie-j) [reply packages/manager/apps/cda/package.json]: @radireddy no the necessary fixes has been made but we'll now need to use `^9.0.0` so the module can build as `^8.0.0` won't match any module now

**Comment by marie-j** [comment]
Hello @radireddy, 
Could you please resolve existing conflicts in order to be able to merge this PR ? 
Thanks

**Comment by JeremyDec** [comment]
Hello @radireddy,

I see the label `translation-required` present => is there a JIRA ticket associated with this demand ?

**Comment by ovh-cds** [comment]
CDS Report installation#16125.0 ✘  
* Stage 1
  * Install ✘

**Comment by ovh-cds** [comment]
CDS Report installation#16128.0 ✘  
* Stage 1
  * Install ✘

File: packages/manager/modules/cda/package.json [hunk 4]
```diff
@@ -0,0 +1,33 @@
+{
+  "name": "@ovh-ux/manager-cda",
+  "version": "0.0.0",
```
Comment (cbourgois) [inline packages/manager/modules/cda/package.json:3]: The commit that introduces this package should contains a `BREAKING CHANGE`

**Comment by ovh-cds** [comment]
CDS Report installation#16186.0 ✘  
* Stage 1
  * Install ✘

**Comment by ovh-cds** [comment]
CDS Report installation#16189.0 ✘  
* Stage 1
  * Install ✘

**Review by varun257 — APPROVED** [review]

**Comment by ovh-cds** [comment]
CDS Report installation#16259.0 ✘  
* Stage 1
  * Install ✘

**Comment by ovh-cds** [comment]
CDS Report installation#16259.1 ✘  
* Stage 1
  * Install ✘

**Review by cbourgois — CHANGES_REQUESTED** [review]
Seems not up to date (uses `ovh-ui-angular`)

File: packages/manager/apps/cda/package.json [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+    "angular-translate": "^2.18.1",
+    "angular-translate-loader-pluggable": "^1.3.1",
+    "angular-ui-bootstrap": "1.3.3",
+    "bootstrap": "~3.3.7",
+    "chart.js": "chartjs/Chart.js#^2.0",
+    "flatpickr": "~4.5.2",
+    "jquery": "^2.1.3",
+    "moment": "^2.19",
+    "ovh-api-services": "^9.41.1",
+    "ovh-ui-angular": "^3.15.1",
```
Comment (cbourgois) [inline packages/manager/apps/cda/package.json:59]: We don't use `ovh-ui-angular` anymore, so I think we should rebase/update this PR.

**Review by cbourgois — CHANGES_REQUESTED** [review]

**Comment by cbourgois** [comment]
Error on installation: 
```
warning Lockfile has incorrect entry for "@ovh-ux/manager-core@^8.0.0 || ^9.0.0". Ignoring it.
Couldn't find any versions for "@ovh-ux/manager-core" that matches "^8.0.0 || ^9.0.0"

```

**Comment by radireddy** [comment]
> Error on installation:
> 
> ```
> warning Lockfile has incorrect entry for "@ovh-ux/manager-core@^8.0.0 || ^9.0.0". Ignoring it.
> Couldn't find any versions for "@ovh-ux/manager-core" that matches "^8.0.0 || ^9.0.0"
> ```

thanks a lot. fixed it

**Comment by radireddy** [comment]
> packages/manager/apps/cloud/client/app/app.js#L112-L118

added the redirection for CDA

**Comment by ovh-cds** [comment]
CDS Report translator-push#13139.0 ✘  
* 
  * push ✘

**Comment by ovh-cds** [comment]
CDS Report translator-push#13379.0 ✘  
* 
  * push ✘

**Review by antleblanc — CHANGES_REQUESTED** [review]
@radireddy To validate changes against PR checks (build/test), could you please rebase on `develop` branch?

Thank you!

**Comment by antleblanc** [comment]
There are some conflicts:
- packages/manager/apps/cloud/client/app/app.js
- packages/manager/apps/dedicated/client/app/app.js

In addition, `test` pipeline reports an error.

Could you please have a look in order to be able to merge this Pull Request?
Thank you!

**Comment by antleblanc** [comment]
ping @radireddy

**Comment by antleblanc** [comment]
Rewriting history in order to fixup some `package.json` files.

**Review by antleblanc — APPROVED** [review]

**Review by antleblanc — APPROVED** [review]

**Pull request merged**